# Задание 2

Два лучника стреляют в мишень.
Когда они независимо стреляют, тренируются, то первый из них попадает в мишень с вероятностью *70%*, а второй - *55%*. 
Теперь они выстрелили вместе, в результате в мишени торчит одна стрела.

Вопрос: какова вероятность, что **попал первый лучник**, а какова - что **попал второй**?

---
## Решение

Известны вероятности независимых событий:
 - вероятность попадания первого лучника `p_arch1 = 0.70`;
 - вероятность попадания второго лучника `p_arch2 = 0.55`;

В результате испытания - произошло одно из них.

Опредлеим вероятности промахов по мишени у лучников:

In [1]:
p_arch1 = 0.70
p_arch2 = 0.55

q_arch1 = 1 - p_arch1
q_arch2 = 1 - p_arch2
print(f'- вероятность промаха первого лучника {q_arch1:0.3f}')
print(f'- вероятность промаха второго лучника {q_arch2:0.3f}')

- вероятность промаха первого лучника 0.300
- вероятность промаха второго лучника 0.450


Для дальнейшего расчёта введём обозначения:
 - `A` первый лучник попал в цель, т.е. `p(A) = p_arch1`;
 - `B` второй лучник попал в цель, т.е. `p(B) = p_arch2`;
 - `not_A` первый лучник промазал, т.е. `p(not_A) = q_arch1`;
 - `not_B` второй лучник промазал, т.е. `p(not_B) = q_arch2`;

В задании обозначено событие `Y` - при двух выстрелах попадает только один из лучников.
Это возможно, когда:
 - первый лучник попал в цель и второй промазал то есть `A * not_B` или `p_arch1 * q_arch2`
 - второй лучник попал в цель и первый промазал то есть `B * not_A` или `p_arch2 * q_arch1`

Так как других вариантов для получения одного попадания нет, а эти два варианта - несовместные (они не могут произойти одновременно), то по теореме сложения вероятностей несовместных событий определим вероятность события `Y`

`p_Y = A * not_B * + not_A * B`

или

`p_Y = p_arch1 * q_arch2 + q_arch1 * p_arch2`

In [2]:
p_Y = p_arch1 * q_arch2 + q_arch1 * p_arch2
print(f'Вероятность, что при двух выстрелах попадает только один из лучников: {p_Y:0.3f}')

Вероятность, что при двух выстрелах попадает только один из лучников: 0.480


Для определениия гипотезы попадания первого лучника - `H1`, второго лучника - `H2` воспользуемся формулой Байеса:

In [3]:
p_A = (q_arch2 * p_arch1) / (p_Y)
print(f'Вероятность, что в мишени находится стрела выпущенная первым лучником: {p_A:0.3f}')

Вероятность, что в мишени находится стрела выпущенная первым лучником: 0.656


In [4]:
p_B = (q_arch1 * p_arch2) / (p_Y)
print(f'Вероятность, что в мишени находится стрела выпущенная первым лучником: {p_B:0.3f}')

Вероятность, что в мишени находится стрела выпущенная первым лучником: 0.344


Так как события `A` и `B` составляют полную группу попарно несовместных событий. Тогда их сумма – событие достоверное, а вероятность достоверного события равна `1`. Выполним проверку условия `p_A + p_B == 1`:

In [5]:
print(f'Проверка условия p_A + p_B == 1: {p_A + p_B:0.3f}')

Проверка условия p_A + p_B == 1: 1.000


## Ответ

Вероятность озвученного в задании события, что при двух выстрелах попадает только один из лучников: 0.480

Вероятность, что в мишени находится стрела выпущенная первым лучником: 0.656

Вероятность, что в мишени находится стрела второго лучника: 0.344
